src/webgpu/util/conversion.ts

```diff
@@ -1112,6 +1112,35 @@ export function scalarTypeOf(ty: Type): ScalarType {
   throw new Error(`unhandled type ${ty}`);
 }
 
+/** @returns the concrete type of the given Type.
```

Comment:
Also it might be worth mentioning that this is the _implicit_ concretization type. If the receiver of the expression has an explicit type, then that has to be used.

Reply:
Done.